### ISSUE TITLE
no basic auth credentials

### ISSUE THREAD

**Issue by XI2171-sumitkadian** [aws-actions/amazon-ecr-login#337]
Some time when pipeline fails with the below error:
`no basic auth credentials`, 
it  does not happens regularly but  in every 10-15 days this issue is occurring.

Pipeline steps:

      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
         aws-region: ${{env.REGION}}
         role-to-assume: ${{env.ASSUME_ROLE}}
         role-duration-seconds: 1200
         role-skip-session-tagging: true
              

      - name: Login to Amazon ECR 
        id: login-ecr
        uses: aws-actions/amazon-ecr-login@v1
        
      - name: Build, tag, and push image to Amazon ECR
        env:          
          ECR_URL: ${{ steps.login-ecr.outputs.registry }}
        run: |
          docker build -t $ECR_URL/$ECR_REPO:$IMAGE_TAG .
          docker push $ECR_URL/$ECR_REPO:$IMAGE_TAG

**Comment by rahul799** [comment]
Facing the same issue too.

**Comment by RobinFrcd** [comment]
Same issue when running multiple pushes on different actions on the same runner instance.

**Comment by arjraman** [comment]
Could you paste the exact error from your logs here, covering any sensitive information?

**Comment by RobinFrcd** [comment]
Nothing really more useful than OP provided sadly :/

```
The push refers to repository [***.dkr.ecr.eu-west-3.amazonaws.com/***]
00ae2f02e451: Preparing
e9857ac559f9: Preparing
a70a5126b72f: Preparing
4cdb6e4a61: Preparing
d8009a0aa7: Preparing
2421911654: Preparing
435699713645: Preparing
2231911654: Waiting
4b4b8ae407af: Preparing
6e8b10052d: Preparing
435699713645: Waiting
0d7f412381ae: Preparing
954edfb465a4: Preparing
47ad77: Preparing
4b4b8ae407af: Waiting
no basic auth credentials
Error: Process completed with exit code 1.
```

**Comment by arjraman** [comment]
Could it be any of the issues mentioned here: https://docs.aws.amazon.com/AmazonECR/latest/userguide/common-errors-docker.html#error-403?

**Comment by RobinFrcd** [comment]
Pretty sure it's not:
- `You have authenticated to a different region`: Always using the same region, eu-west-3
- `You have authenticated to push to a repository you don't have permissions for`: If I retry a few seconds later, it works so the runner has the right permission
- `Your token has expired`: I always do the ecr login before trying to push, so no
- `Bug in wincred credential manager`: The runner is on Linux

**Comment by nerdeveloper** [comment]
@RobinFrcd what was your workaround concerning this issue

**Comment by nerdeveloper** [comment]
OMG! I just knew it. but I thought you'd be able to make it work without having to do that. The solution to this is to increase costs. 

@arjraman, any time to fix this issue

**Comment by nerdeveloper** [comment]
This is my workaround. I used the native docker login; you can use this if you choose to have multiple runners in one VM pending when this issue is fixed. 

```bash
      - name: Login into ECR 
        run: echo ::set-output name=logged-in::$(aws ecr get-login-password --region us-west-1 | docker login --username AWS --password-stdin [private ECR repo URL])
```

**Comment by zMynxx** [comment]
Still experiencing this issue on my self-hosted runner, a yeah later.
Any plan to find a proper solution?

**Comment by RobinFrcd** [comment]
@zMynxx If I remember right, I fixed this issue with a `skip-logout: true`

```
      - name: Login to Amazon ECR
        id: login-ecr
        uses: aws-actions/amazon-ecr-login@v2
        with:
          skip-logout: true
```

**Comment by zMynxx** [comment]
> @zMynxx If I remember right, I fixed this issue with a `skip-logout: true`
> 
> ```
>       - name: Login to Amazon ECR
>         id: login-ecr
>         uses: aws-actions/amazon-ecr-login@v2
>         with:
>           skip-logout: true
> ```

Yeah it's all good, I asked bacause I could this [this issue](https://github.com/aws-actions/amazon-ecr-login/issues/623) still open and no responses. I ended up trying this flag and it LGTM
